packages/components/src/slot-fill/use-slot.ts

```diff
@@ -8,16 +12,17 @@ import { useContext, useSyncExternalStore } from '@wordpress/element';
  * Internal dependencies
  */
 import SlotFillContext from './context';
+import type { SlotKey } from './types';
 
 /**
  * React hook returning the active slot given a name.
  *
  * @param {string} name Slot name.
- * @return {Object} Slot object.
+ * @return {Component|undefined} Slot object.
  */
-const useSlot = ( name ) => {
+const useSlot = ( name: SlotKey ) => {
 	const { getSlot, subscribe } = useContext( SlotFillContext );
-	return useSyncExternalStore(
+	return useSyncExternalStore< Component | undefined >(
```

Comment:
It's not immediately clear why `Component | undefined ` is necessary here. Perhaps it would be more explicit if we pass it as another, separately defined type to `getSlot()`?

Reply:
This change was unnecessary in the first place because type inference is fully functional in this area. It will be removed.